### PR TITLE
Add i18n support

### LINKED ISSUE #1: Add i18n support
Pipelines component of Kubeflow UI must be available in both official languages.

### PR DESCRIPTION
**Pull request by brendangadd**
Closes #1

### CODE REVIEW

File: frontend/src/pages/AllRunsList.tsx
Comment (wg102) [inline frontend/src/pages/AllRunsList.tsx:107]: here both common and experiments translation files are set to t, but only the common part seems to be used by the element. Is the other used by a child? 
Was the purpose to sent t to RunList through the props?

File: frontend/src/i18n.ts
Comment (wg102) [inline frontend/src/i18n.ts:7]: Is this file mandatory , or was it created for our own purposes? I am asking because it is used directly in a few places, and the name could lead to confusion.
I noticed a few places where it is used, and it can also be used for the missing translation points 1-4. Or using Translation (https://react.i18next.com/latest/translation-render-prop)

File: frontend/src/pages/NewRun.tsx
Comment (wg102) [inline frontend/src/pages/NewRun.tsx:1148]: This might happen in other places then here, but there should be a way in i18next to set make in one label, and insert the values in it. Might be https://www.i18next.com/translation-function/nesting#passing-options-to-nestings or https://www.i18next.com/translation-function/interpolation

File: frontend/public/locales/en/common.json
Comment (wg102) [inline frontend/public/locales/en/common.json:108]: This might be happening in more places, but translations such as these are hard to do because the 'sentence' might  have a different structure in other language. If possible, need to find a way to have one label, where you have attributes for the values.  (see NewRun.tsx comment)

File: frontend/src/pages/NewRun.tsx
Comment (wg102) [inline frontend/src/pages/NewRun.tsx:909]: Same comment as before, error message could probably have a way to take argument, and therefore include the embeddedRunId in it's message directly.

File: frontend/public/locales/en/common.json
Comment (wg102) [inline frontend/public/locales/en/common.json:1]: Some of the labels, especially in the common file, could potentially be easier to read if we format them like https://www.i18next.com/translation-function/essentials#accessing-keys.
I did not test if it was applicable.

File: frontend/public/locales/en/experiments.json
Comment (wg102) [inline frontend/public/locales/en/experiments.json:3]: There should be a way to avoid split messages. 
(Recurrent thing happening)

File: frontend/src/TestUtils.tsx
Comment (wg102) [inline frontend/src/TestUtils.tsx:30]: Not used?

**Review by wg102 — COMMENTED** [review]
I noticed the uses of both withTranslation and useTranslation, depending on the type of file is was (class based or component based). 
There is also the creation of the i18n.tsx file. Which seems to work great and could be used for some of the missing translations. The other way would be to use the [Translation](https://react.i18next.com/latest/translation-render-prop). I want to make sure of how the file works, because it's use seems cleaner than the use of the suggested one from i18next.

The comments are a bit repetitive, and are a first pass:

1. Some of the translation will be really hard to do, because they are split in parts. In usual translation tools, there should be a way to specify data. https://www.i18next.com/translation-function/nesting#passing-options-to-nestings or https://www.i18next.com/translation-function/interpolation
2. the i18n file, and the name

**Comment by Nancy-Badran** [comment]
To load the translation of Getting started page we can use the Trans Component https://react.i18next.com/latest/trans-component , the idea is find the right way to write the JSON file destinated to load the translated Lines. 
take a look on frontend/source/pages/Gettingstarted.tsx.

**Comment by Nancy-Badran** [comment]
And to translate the enum , it seem that we  can directly access t on the i18n instance: https://www.i18next.com/overview/api#t, I did it as the same way but  I also had the error below:
 `Computed values are not permitted in an enum with string valued members` . I think We must fill the enum in the callback as suggested in the issue https://github.com/i18next/react-i18next/issues/479.

**Comment by Nancy-Badran** [comment]
Added Translation to getting started

File: frontend/public/locales/en/common.json
Comment (wg102) [inline frontend/public/locales/en/common.json:194]: 1. Please confirm if there is any other way than a line by line translation.
2. May want to investigate if we should even have that page
3. We may want to put these in a separate json file, or at least not under the common one.

File: frontend/src/i18n.ts
Comment (wg102) [inline frontend/src/i18n.ts:20]: remove new lines

File: frontend/src/pages/GettingStarted.tsx
Comment (wg102) [inline frontend/src/pages/GettingStarted.tsx]: Are these values still in use?

File: frontend/src/pages/GettingStarted.tsx
Comment (wg102) [inline frontend/src/pages/GettingStarted.tsx:103]: What value is in need of passing here?

File: frontend/public/locales/en/common.json
Comment (wg102) [reply frontend/public/locales/en/common.json:194]: There seems to be a 'one line' way of having json 
This is just a quick test, but using `\n`
```
"key2": " **Demos234** say: {{val}} <br/> \n ## Build your own pipeline with \n * TensorFlow Extended (TFX) [SDK](https://www.tensorflow.org/tfx/guide) with end-to-end ML Pipeline Template ([Open TF 2.1 Notebook](https://console.cloud.google.com/mlengine/notebooks/deploy-notebook?q=download_url%3Dhttps%253A%252F%252Fraw.githubusercontent.com%252Ftensorflow%252Ftfx%252Fmaster%252Fdocs%252Ftutorials%252Ftfx%252Ftemplate.ipynb)) \n      * Kubeflow Pipelines [SDK](https://www.kubeflow.org/docs/pipelines/sdk/) \n    <br/>"
```

File: frontend/src/pages/GettingStarted.tsx
Comment (Nancy-Badran) [reply frontend/src/pages/GettingStarted.tsx]: I believe yes, in the  public render(): JSX.Element {...

File: frontend/src/pages/NewRun.tsx
Comment (Nancy-Badran) [reply frontend/src/pages/NewRun.tsx:909]: {      
    "key1": "{${embeddedRunId}}",
     "key2": "Error: failed to retrieve the specified run:{{val}}"
}
i18next.t('key2', {val: '$t(key1)'});
Need to find a way to pass the value of embeddedRunId, didnt worked in this way. And if we decide to use the nested or interpolation function, must change here and many mores places.

File: frontend/public/locales/en/common.json
Comment (Nancy-Badran) [reply frontend/public/locales/en/common.json:194]: I will create a gettingstarted.json to this page and try to load it in one paragraph.

**Comment by Nancy-Badran** [comment]
Bug: In the getting started page , the translated text of the page title is loaded if we pass through the pipeline page to the getting started page, but if we go forward to localhost:3000/#/start , the translation don't appear to be loaded.

File: frontend/src/lib/Buttons.ts
Comment (wg102) [inline frontend/src/lib/Buttons.ts:883]: IMPORTANT: Are we missing this whole translation?

File: frontend/src/lib/TriggerUtils.ts
Comment (wg102) [inline frontend/src/lib/TriggerUtils.ts:325]: Many new lines in this file

File: frontend/src/lib/TriggerUtils.ts
Comment (wg102) [inline frontend/src/lib/TriggerUtils.ts:321]: IMPORTANT lines 291 to 314 seem to include untranslated text

File: frontend/src/pages/Compare.tsx
Comment (wg102) [inline frontend/src/pages/Compare.tsx:342]: If the line is commented I only assume it was a test. Please remove it, or use the line appropriate. I think i18n is declared but not used

File: frontend/src/pages/Compare.tsx
Comment (wg102) [inline frontend/src/pages/Compare.tsx:45]: I don't see where this import of i18n is used

**Review by wg102 — CHANGES_REQUESTED** [review]
General comments:
- I used keyword BUG to flag features in need of fix
- I used keyword IMPORTANT to flag things we might have missed
- Almost all json files are missing the newline at the end of the file
- Most comment on the json English files also apply to the French ones
- Many files have superfluous new lines through them (visible on code comparison in the PR) 
- I noticed that in "ResourceSelector.tsx", the logger.error messages are translated, but that was not the case in most of other files. Is this something we want, translation in logger? If so, do we need to do it in all other places as well?

File: frontend/src/lib/TriggerUtils.ts
Comment (Nancy-Badran) [reply frontend/src/lib/TriggerUtils.ts]: Couldn't find where it is used in the UI

File: frontend/src/components/Trigger.tsx
Comment (Nancy-Badran) [reply frontend/src/components/Trigger.tsx]: the use of (s) could be a solution. 
I used here but we still can find the origin of the pb .

File: frontend/src/components/Trigger.tsx
Comment (wg102) [inline frontend/src/components/Trigger.tsx:306]: Month in French will show as "Mois(s)"

File: frontend/src/components/Trigger.tsx
Comment (wg102) [inline frontend/src/components/Trigger.tsx:315]: I am wondering if this code will behave as expected if the language is in French. Same hesitancy applies to around line 115 for other enum uses. `type: parsedTrigger.type ?? TriggerType.INTERVALED`

**Review by wg102 — COMMENTED** [review]
As discussed, the last few things that need to be checked before approval.
These and the test cases.

File: frontend/src/components/Trigger.tsx
Comment (brendangadd) [inline frontend/src/components/Trigger.tsx:329]: This needs i18n.

File: frontend/src/components/Trigger.tsx
Comment (Nancy-Badran) [reply frontend/src/components/Trigger.tsx:315]: @brendangadd

File: frontend/src/components/Trigger.tsx
Comment (Nancy-Badran) [reply frontend/src/components/Trigger.tsx:306]: I think the plural concept doesn't works because  even if we choose 1 week in the  'real site' h[ttps://kubeflow.covid.cloud.statcan.ca/_/pipeline/#/runs/new?experimentId= ](url) it put a '  s ' . So I suggest always have a s in that enum

Comment (Nancy-Badran) [reply frontend/src/components/Trigger.tsx:306]: image

File: frontend/src/pages/ResourceSelector.test.tsx
Comment (brendangadd) [inline frontend/src/pages/ResourceSelector.test.tsx:157]: Remove

File: package.json
Comment (brendangadd) [inline package.json:74]: Surprised to see these additions. If `@types/jest` is required for making TypeScript tooling happy, then cool, but it's probably a dev dependency in that case.

Redux seems out of place.

**Review by brendangadd — COMMENTED** [review]
Wow, a lot of work went into this. Great job!

I kept my comments ultra short and to the point due to the size of the review, so please ignore any poor tone trying to hide in there. :)

File: frontend/src/components/PlotCard.tsx
Comment (Nancy-Badran) [reply frontend/src/components/PlotCard.tsx:108]: Use clone instead ?

File: frontend/src/pages/ArtifactList.tsx
Comment (Nancy-Badran) [reply frontend/src/pages/ArtifactList.tsx:128]: We didn't wrapped, it is referenced out in many other components, and we tried to export it with  `{withRef :true}` option but it didn't worked properly..

File: frontend/public/locales/fr/experiments.json
Comment (brendangadd) [inline frontend/public/locales/fr/experiments.json:167]: Whitespace (don't rely on leading or trailing whitespace for messages)

File: frontend/public/locales/fr/home.json
Comment (brendangadd) [inline frontend/public/locales/fr/home.json:4]: Newline

File: frontend/src/TestUtils.tsx
Comment (brendangadd) [inline frontend/src/TestUtils.tsx:75]: Whitespace

File: frontend/src/TestUtils.tsx
Comment (brendangadd) [inline frontend/src/TestUtils.tsx:94]: Whitespace

File: frontend/src/components/CustomTable.tsx
Comment (brendangadd) [inline frontend/src/components/CustomTable.tsx:211]: * Whitespace
* Why was the export moved down?

File: frontend/src/components/ExperimentList.tsx
Comment (brendangadd) [inline frontend/src/components/ExperimentList.tsx:39]: Whitespace

File: frontend/src/components/ExperimentList.tsx
Comment (brendangadd) [inline frontend/src/components/ExperimentList.tsx:47]: Whitespace

File: frontend/src/components/Router.test.tsx
Comment (brendangadd) [inline frontend/src/components/Router.test.tsx:40]: `{}` is not `TFunction`.

File: frontend/src/pages/ExperimentList.test.tsx
Comment (brendangadd) [inline frontend/src/pages/ExperimentList.test.tsx:526]: Spelling, formatting, whitespace

File: frontend/src/pages/GettingStarted.tsx
Comment (brendangadd) [inline frontend/src/pages/GettingStarted.tsx:99]: Awesome.

File: frontend/src/pages/ResourceSelector.test.tsx
Comment (brendangadd) [inline frontend/src/pages/ResourceSelector.test.tsx:90]: Remove

**Pull request merged**
